src/compiler/checker.ts

```diff
@@ -3197,10 +3221,22 @@ namespace ts {
                     if (!ignoreErrors) {
                         const namespaceName = getFullyQualifiedName(namespace);
                         const declarationName = declarationNameToString(right);
-                        const suggestion = getSuggestedSymbolForNonexistentModule(right, namespace);
-                        suggestion ?
-                            error(right, Diagnostics._0_has_no_exported_member_named_1_Did_you_mean_2, namespaceName, declarationName, symbolToString(suggestion)) :
+                        const suggestionForNonexistentModule = getSuggestedSymbolForNonexistentModule(right, namespace);
+                        const exportedTypeSymbol = getMergedSymbol(getSymbol(getExportsOfSymbol(namespace), right.escapedText, SymbolFlags.Type));
+                        const fullQualifiedName = isQualifiedName(name) && getFullQualifiedName(name);
+                        const canSuggestTypeof = fullQualifiedName && !isTypeOfExpression(fullQualifiedName.parent) && tryGetQualifiedNameAsValue(fullQualifiedName);
+                        if (suggestionForNonexistentModule) {
+                            error(right, Diagnostics._0_has_no_exported_member_named_1_Did_you_mean_2, namespaceName, declarationName, symbolToString(suggestionForNonexistentModule));
+                        }
+                        else if (canSuggestTypeof) {
+                            error(name, Diagnostics._0_refers_to_a_value_but_is_being_used_as_a_type_here_Did_you_mean_typeof_0, entityNameToString(fullQualifiedName as QualifiedName));

```

Comment:
Haven't tried TypeScript 4.4, so I wrote `fullQualifiedName as QualifiedName`. After upgrading vscode, I find it is unnecessary. The CFA is really impressive, kudos to you guys!